resources/rds-dbclusterparametergroups.go

```diff
@@ -0,0 +1,56 @@
+package resources
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+)
+
+type RDSDBClusterParameterGroup struct {
+	svc  *rds.RDS
+	name *string
+}
+
+func (n *RDSNuke) ListClusterParameterGroups() ([]Resource, error) {
+	params := &rds.DescribeDBClusterParameterGroupsInput{MaxRecords: aws.Int64(100)}
+	resp, err := n.Service.DescribeDBClusterParameterGroups(params)
+	if err != nil {
+		return nil, err
+	}
+	var resources []Resource
+	for _, parametergroup := range resp.DBClusterParameterGroups {
+		resources = append(resources, &RDSDBClusterParameterGroup{
+			svc:  n.Service,
+			name: parametergroup.DBClusterParameterGroupName,
+		})
+
+	}
+
+	return resources, nil
+}
+
+func (i *RDSDBClusterParameterGroup) Filter() error {
+	if strings.HasPrefix(*i.name, "default.") {
+		return fmt.Errorf("Cannot delete default parameter group")
+	}
+	return nil
+}
+
+func (i *RDSDBClusterParameterGroup) Remove() error {
+	params := &rds.DeleteDBClusterParameterGroupInput{
+		DBClusterParameterGroupName: i.name,
+	}
+
+	_, err := i.svc.DeleteDBClusterParameterGroup(params)
+	if err != nil {
```

Comment:
Couldn't you just skip this `if` and directly: `return err`

Reply:
IMHO the logic is more understandable and separated like this...
first: error handling
then: making the function return what is should

Reply:
IMO it looks like unnecessary code :>

Reply:
You could, but then you are implying that whoever looks at the code afterwards understands that `err` defaults to `nil`. The if statement makes it more explicit and easier to read what's going on. Also, this is how that error is handled across all the classes in aws-nuke where it's applicable, so it is following a common pattern in the program.

Reply:
Decision?

Reply:
Your call

Reply:
I will leave it in for consistency's sake. If this is a big enough problem then it deserves its own pull request and that if statement should be removed everywhere across the whole application.

Reply:
Very much agree. Greatly in favor of consistency.